client-misk-jooq/src/main/kotlin/app/cash/backfila/client/misk/jooq/internal/Validator.kt

```diff
@@ -0,0 +1,34 @@
+package app.cash.backfila.client.misk.jooq.internal
+
+import app.cash.backfila.protos.clientservice.KeyRange
+import com.squareup.backfila.client.base.jooq.ByteStringSerializer
+import misk.exceptions.BadRequestException
+
+
+object Validator {
+  /**
+   * Validates a KeyRange by verifying that start or end can be parsed by the given serializer if
+   * they are non-null.
+   *
+   * @throws BadRequestException if the KeyRange is invalid
+   */
+  fun validateRange(range: KeyRange?, serializer: ByteStringSerializer<*>) {
+    if (range == null) {
+      return
+    }
+    try {
+      if (range.start != null) {
+        serializer.fromByteString(range.start)
+      }
+    } catch (e: NumberFormatException) {
+      throw BadRequestException("Start of range must be a number")
```

Comment:
Do we really need to only accept number primary keys?

Reply:
No we don't. I'll change that to catch a generic exception, instead. Cause it should fail if we can't serialise / deserialise it

Reply:
I had to remove this altogether as compound keys cant have this validation anyway.